tests/projects/run_tests.py

```diff
@@ -0,0 +1,217 @@
+    elif verbose:
+        print(err)
+
+    return 0, "Success"
+
+def clean_helper(path_to_test: string):
+    shutil.rmtree(os.path.join(path_to_test, "base", "build"))
+    shutil.rmtree(os.path.join(path_to_test, "downstream", "build"))
+    shutil.rmtree(os.path.join(path_to_test, "tmp_install_dir"))
+
+def run_test(path_to_test: string, blt_source_dir: string, host_config: string, verbose=False, clean=False):
+    """ Run test, using a yaml to specify CMake arguments """
+    # CMake, Build and install base
+    code, err = cmake_build_project(path_to_test, blt_source_dir, host_config, True, verbose)
+    if code:
+        if clean:
+            shutil.rmtree(os.path.join(path_to_test, "base", "build"))
+        return code, err
+    # CMake, build downstream
+    code, err = cmake_build_project(path_to_test, blt_source_dir, host_config, False, verbose)
+
+    if clean:
+        clean_helper(path_to_test)
+
+    if code:
+        return code, err
+    
+    return 0, "Test {0} passed".format(path_to_test)
+
+def parse_args():
+    "Parses args from command line"
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--host-config",
+                      dest="host-config",
+                      default=None,
+                      help="Host config file to be used by all test projects.")
+
+    # Where to find BLT
+    parser.add_argument("--blt-source-dir",
+                      dest="blt-source-dir",
+                      default=None,
+                      help="Path to BLT source to be used by all test projects.")
+
+    # Verbose mode: useful for debugging.
+    parser.add_argument("--verbose",
+                      action='store_true',
+                      dest="verbose",
+                      help="Print all stdout and stderr from running tests.")
+
+    # Specify a subset of tests to run.  Useful for debugging
+    parser.add_argument("--run-test",
+                      default=None,
+                      dest="run-test",
+                      help="Comma delimited list of tests to run.  Test names must be a subset of the "
+                            "list of directories inside test. Only run tests specified.")
+
+    # Specify whether to clean build and install directories
+    parser.add_argument("--clean",
+                      action='store_true',
+                      help="Remove build and install paths from test directories.")
+
+    args, extra_args = parser.parse_known_args()
+    args = vars(args)
+    
+    if not args["blt-source-dir"]:
+        print("[ERROR: Required command line argument, 'blt-source-dir', was not provided.]")
+        return None
+
+    if args["host-config"] is not None and not os.path.exists(args["host-config"]):
+        print("ERROR: Host config file {0} specified, but file does not exist.".format(args["host-config"]))
+        return None
+
+    if args["run-test"] is not None:
+        all_tests = set(os.listdir(os.path.relpath("test")))

```

Comment:
```suggestion
        all_tests = set(os.listdir(os.path.relpath(".")))

```

I think this is right..